src/api/app/models/event/build.rb

```diff
@@ -54,5 +63,21 @@ def my_message_id
       mid = Digest::MD5.hexdigest("#{payload['project']}-#{payload['package']}-#{payload['repository']}-#{md5}")
       "<build-#{mid}@#{self.class.message_domain}>"
     end
+
+    # rubocop:disable Metrics/CyclomaticComplexity
+    def maintained_project_name
+      maintenance_project = Project.get_maintenance_project
+      return unless maintenance_project
+
+      return maintenance_project.name if payload['project'].start_with?(maintenance_project.name)
+
+      # maintained projects and their sub-projects
+      return unless maintenance_project.maintained_project_names.any? { |name| payload['project'].start_with?(name) }
+
+      maintenance_project.maintained_project_names.sort_by(&:length).reverse_each do |name|
+        return name if name.in?(payload['project'])
+      end
+    end
+    # rubocop:enable Metrics/CyclomaticComplexity
```

Comment:
```suggestion
    def maintained_project_name
      maintenance_project = Project.get_maintenance_project
      return unless maintenance_project

      return maintenance_project.name if payload['project'].start_with?(maintenance_project.name)

      # maintained projects and their sub-projects
      return unless maintenance_project.maintained_project_names.any? { |name| payload['project'].start_with?(name) }

      maintenance_project.maintained_project_names.sort_by(&:length).reverse.find { |name| name.in?(payload['project']) }
    end
```

You can use this to make it simpler

Reply:
hm, where is the difference exactly?

Reply:
Combining conditions and loops makes thing harder to understand

Reply:
So this: `maintenance_project.maintained_project_names.sort_by(&:length).reverse.find { |name| name.in?(payload['project']) }`